### ISSUE TITLE
Query about the point cloud RGB color

### ISSUE THREAD

**Issue by RayYoh** [scannetpp/scannetpp#72]
Hi authors,

Recently, I am preprocessing this dataset, but I find one minor difference about the point cloud color.
Some previous preprocess code for ScanNetv2 directly use package `plyfile` to read the mesh file to get the coord and the color, like below:
```
def read_plymesh(filepath):
    """Read ply file and return it as numpy array. Returns None if emtpy."""
    with open(filepath, "rb") as f:
        plydata = plyfile.PlyData.read(f)
    if plydata.elements:
        vertices = pd.DataFrame(plydata["vertex"].data).values
        faces = np.stack(plydata["face"].data["vertex_indices"], axis=0)
        return vertices, faces
```
And I have noticed that we need to sample points on the mesh to get the point cloud. But I am confused about the color of the point cloud [here](https://github.com/scannetpp/scannetpp/blob/34a10a66503c77cb924db4ef618993d6ffe0d9c0/semantic/transforms/mesh.py#L196). And I have printed the processed color of one scene (41b00feddb). It is:
```
processed colors:  [[0.40814825 0.41282975 0.41206982]
 [0.31537095 0.31422019 0.31495419]
 [0.43031899 0.42329089 0.41234123]
 ...
 [0.63921569 0.63856839 0.63809172]
 [0.67151643 0.66645789 0.67481222]
 [0.66787996 0.67665181 0.67598751]]
```
It seems no relation with the RGB color from 0-255, my question is what these data mean?
And I also print the color data of the `read_plymesh`, its max and min are 255 and 0.

I'd like to ask can I use this ndx [here](https://github.com/scannetpp/scannetpp/blob/34a10a66503c77cb924db4ef618993d6ffe0d9c0/semantic/transforms/mesh.py#L221) to select the color data read by `read_plymesn`?

**Comment by RayYoh** [comment]
> Hi authors,
> 
> Recently, I am preprocessing this dataset, but I find one minor difference about the point cloud color. Some previous preprocess code for ScanNetv2 directly use package `plyfile` to read the mesh file to get the coord and the color, like below:
> 
> ```
> def read_plymesh(filepath):
>     """Read ply file and return it as numpy array. Returns None if emtpy."""
>     with open(filepath, "rb") as f:
>         plydata = plyfile.PlyData.read(f)
>     if plydata.elements:
>         vertices = pd.DataFrame(plydata["vertex"].data).values
>         faces = np.stack(plydata["face"].data["vertex_indices"], axis=0)
>         return vertices, faces
> ```
> 
> And I have noticed that we need to sample points on the mesh to get the point cloud. But I am confused about the color of the point cloud [here](https://github.com/scannetpp/scannetpp/blob/34a10a66503c77cb924db4ef618993d6ffe0d9c0/semantic/transforms/mesh.py#L196). And I have printed the processed color of one scene (41b00feddb). It is:
> 
> ```
> processed colors:  [[0.40814825 0.41282975 0.41206982]
>  [0.31537095 0.31422019 0.31495419]
>  [0.43031899 0.42329089 0.41234123]
>  ...
>  [0.63921569 0.63856839 0.63809172]
>  [0.67151643 0.66645789 0.67481222]
>  [0.66787996 0.67665181 0.67598751]]
> ```
> 
> It seems no relation with the RGB color from 0-255, my question is what these data mean? And I also print the color data of the `read_plymesh`, its max and min are 255 and 0.
> 
> I'd like to ask can I use this ndx [here](https://github.com/scannetpp/scannetpp/blob/34a10a66503c77cb924db4ef618993d6ffe0d9c0/semantic/transforms/mesh.py#L221) to select the color data read by `read_plymesn`?

Sry, I know that. They are just RGB/255.

**Comment by cy94** [comment]
Yes Open3D uses colors in 0-1, and use the KDTree indices can be used to pick the colors on the visualized points, or you can get them using the semantic labels.

**Comment by RayYoh** [comment]
> Yes Open3D uses colors in 0-1, and use the KDTree indices can be used to pick the colors on the visualized points, or you can get them using the semantic labels.

Hello, Chandan. Thanks for your patient reply to my questions.
Now, I have another confusion. After I prepared the training data, I find that every scene has millions of points (I use the default sample factor [here](https://github.com/scannetpp/scannetpp/blob/34a10a66503c77cb924db4ef618993d6ffe0d9c0/semantic/configs/prepare_training_data.yml.template#L20)). The number of points is very more significant than ScanNetv2, based on my previous code for ScanNetv2, the time of data augmentation will be very long. So, may I ask how you preprocess these points, did you do any sub-sample operations during your training? And is there any possibility of sharing your implementation details? 
![image](https://github.com/scannetpp/scannetpp/assets/35106336/1ee3a684-a21e-457c-a5f1-3b4b8df65315)

I find that the benchmark now is different from what you introduced in the paper (for the semantic classes and instance classes).

**Comment by cy94** [comment]
You can reduce the points right now using the sample factor if you like, or you can just train on the raw sampled points and voxelize them which will reduce the number of points, and then apply the augmentations on the voxels. For the baselines I had also split the scenes into 5mx5m chunks because there are some big scenes which won't fit into a small GPU. I'll try to add that code to our repo, but you could also try implementing it yourself.

**Comment by cy94** [comment]
Yes, the number of classes has been increased in the benchmark as compared to the paper because we had more data.

**Comment by RayYoh** [comment]
> You can reduce the points right now using the sample factor if you like, or you can just train on the raw sampled points and voxelize them which will reduce the number of points, and then apply the augmentations on the voxels. For the baselines I had also split the scenes into 5mx5m chunks because there are some big scenes which won't fit into a small GPU. I'll try to add that code to our repo, but you could also try implementing it yourself.

Thanks for your patient reply.

**Comment by cy94** [comment]
You can try this out https://github.com/scannetpp/scannetpp?tab=readme-ov-file#split-pth-files-into-chunks-for-training

**Comment by RayYoh** [comment]
> You can try this out https://github.com/scannetpp/scannetpp?tab=readme-ov-file#split-pth-files-into-chunks-for-training

Thanks for your generous sharing, Chandan. And thanks for your patience again.

Now, I am just based on this repo [Pointcept](https://github.com/Pointcept/Pointcept/blob/553bf5a1520d7e9326aec4c602f42bf50b76b4c8/configs/scannet/insseg-pointgroup-v1m1-0-spunet-base.py#L93C13-L101C69) to reproduce the result of PointGroup for instance segmentation. Specifically, I use GridSample (grid_size=0.05, same to S3DIS) and SpereCrop to randomly crop 0.8x grids. It can be trained on my devices, but I am not sure whether this config will influence the final performance (it may ignore some small instances?)

And I also plan to use segmentator to get the suerpoints like [here](https://github.com/Karbo123/segmentator), which will significantly reduce computational resources. It also shows better performance on ScanNetv2 for ins seg.

If both do not work, then I will try this (since I haven't understood this one) LoL. 

Thank you again for your help.

**Comment by cy94** [comment]
Sure. The `split_pth` is splitting each scene into 5mx5m chunks and creating new "scenes" from them. The output is many smaller pth files, and a scene list with the corresponding chunk names which is `{scene_id}_{i}`. I used the [Minsu3D](https://github.com/3dlg-hcvc/minsu3d) framework for PointGroup which was quite easy to setup.

**Comment by RayYoh** [comment]
> Sure. The `split_pth` is splitting each scene into 5mx5m chunks and creating new "scenes" from them. The output is many smaller pth files, and a scene list with the corresponding chunk names which is `{scene_id}_{i}`. I used the [Minsu3D](https://github.com/3dlg-hcvc/minsu3d) framework for PointGroup which was quite easy to setup.

Thx. I will try it later.

**Comment by RayYoh** [comment]
> Sure. The `split_pth` is splitting each scene into 5mx5m chunks and creating new "scenes" from them. The output is many smaller pth files, and a scene list with the corresponding chunk names which is `{scene_id}_{i}`. I used the [Minsu3D](https://github.com/3dlg-hcvc/minsu3d) framework for PointGroup which was quite easy to setup.

Hi, Chandan. I have tried this for the train set. But I am still confused [here](https://github.com/scannetpp/scannetpp#split-pth-files-into-chunks-for-training) for the val set. Does it mean we needn't split the val set?

**Comment by cy94** [comment]
Its upto you, since you need less GPU memory during validation you might not need to split it.
